app/controllers/user_controller.rb

```diff
@@ -0,0 +1,68 @@
+class UserController < ApplicationController
+protect_from_forgery :except => :create 
+	def index
+		@users = User.all
+	end
+
+	def new
+		@csrf_token = form_authenticity_token
+	end
+
+	def create
+		begin
+			new_user = User.new(user_params)
+			User.transaction do
```

Comment:
Talk to me and we will work on refactoring this.  This is too much logic in the controller.